### PR TITLE
fix: drop column if not defined in attributes when alter table

### LINKED ISSUE #294: app.model.sync方法不能正确的同步model到数据库
### 复现步骤
通过`javascript model`来更新数据库表结构。
1. 创建普通的egg项目。并且配置数据库链接。使其能够正常链接数据库
2. 创建一个 `user.js`的数据模型
 ```javascript
module.exports = function(app, Model) {
    const { STRING } = Model.DataTypes;
    class User extends Model {
        static attributes = {
            name: STRING,
            password: STRING,
            avatar: STRING,
        }
    }
    return User
  }
```
4. 在egg app.js生命周期中执行model同步数据库
 ```javascript
  async serverDidReady() {
        console.log('同步表结构中...')
        await this.app.model.sync({ alter: true})        
        console.log('表结构同步完成...')
    }
```
5. 数据库表创建数据成功
6. 修改`user.js`为如下
  ```javascript
 module.exports = function(app, Model) {
    const { STRING } = Model.DataTypes;
    class User extends Model {
        static attributes = {
            name: STRING,
            password: STRING,
            // avatar: STRING, // 屏蔽掉avatar属性
        }
    }
    return User
  }
 ```
7. 重新运行egg项目
### 实际结果
数据`user`表中任然存在 `avatar`列。
###  期望结果
数据库表中不存在 `avatar`列表

### 问题溯源
`leoric/src/bone.js` 1020行。执行`compare`比对有问题。
```javascript
Object.defineProperties(this, looseReadonly({
      timestamps,
      table,
      primaryKey,
      columns,
      attributeMap,
      associations,
      tableAlias,
      synchronized: Object.keys(compare(attributes, columnMap)).length === 0,// 此处对比有问题
    }));
```

### PR DESCRIPTION
**Pull request by cyjake**
also (partially) added column comment support to close #294

### CODE REVIEW

File: src/bone.js [hunk 1]
```diff
@@ -50,6 +52,12 @@ function compare(attributes, columnMap) {
     }
   }
 
+  for (const columnName in columnMap) {
+    if (!columnNames.has(columnName)) {
+      diff[columnName] = { remove: true };
```
Comment (barretlee) [inline src/bone.js:57]: `removed`?

File: src/data_types.js [hunk 2]
```diff
@@ -17,17 +17,21 @@ class DataType {
...
+    const params = [];
+    for (let i = 0; i < matches.length; i++) {
+      if (matches[i] != null) params[i] = parseInt(matches[i], 10);
+    }
 
     switch (dataType) {
       case 'varchar':
       case 'char':
-        return new STRING(length);
+        console.log(params);
```
Comment (barretlee) [inline src/data_types.js:33]: remove console.log

File: src/bone.js (hunk 1 above)
Comment (cyjake) [reply src/bone.js:57]: there is `{ modify: true }` already, so... on the other hand, the properties were added in a way that is more like to tell driver what to update not what happened in model attributes.

File: src/data_types.js (hunk 2 above)
Comment (cyjake) [reply src/data_types.js:33]: nice catch

**Review by JimmyDaddy — APPROVED** [review]

**Pull request merged**